### PR TITLE
Update subscription calls

### PR DESCRIPTION
**Pull request by gsvr**
- [x] Remove utils that can be implemented in Python
- [x] Hit new Control Registrations endpoint

### CODE REVIEW

**Comment by gsvr** [comment]
@imsickofmaps Please review.

File: src/chw.js [hunk 1]
```diff
@@ -551,7 +548,8 @@ go.app = function() {
                                     self.user.extra.ussd_sessions = '0';
                                     return Q.all([
                                         self.im.contacts.save(self.user),
-                                        self.im.contacts.save(self.contact)
+                                        self.im.contacts.save(self.contact),
+                                        go.utils.adjust_percentage_registrations(self.im, self.metric_prefix)
```
Comment (gsvr) [inline src/chw.js:552]: @imsickofmaps I moved the `adjust_percentage_registrations` here to minimise `.then` usage, but it seperates the meaning a bit - let me know what you think.

File: src/chw.js [hunk 2]
```diff
@@ -111,9 +111,9 @@ go.app = function() {
                     return creator(name, opts);
 
                 interrupt = false;
-                opts = opts || {};
-                opts.name = name;
-                return self.states.create('states_timed_out', opts);
+                var timeout_opts = opts || {};
```
Comment (gsvr) [inline src/chw.js:114]: @imsickofmaps Trying to avoid creating a global variable here.

File: src/utils.js [hunk 3]
```diff
@@ -524,46 +541,85 @@ go.utils = {
...
+            authority: reg_type,  // 'clinic' | 'chw' | 'personal'
+        };
+        return go.utils
+            .control_api_call("post", null, payload, 'registration/', im)
+            .then(function(post_response) {
+                return go.utils.json_success_fail_metric(im, 'registration_call', post_response);
+            });
+    },
+
+    post_subscription: function(contact, im, metric_prefix, env, opts) {
```
Comment (gsvr) [inline src/utils.js:620]: @imsickofmaps This looks unclear in the review area, I actually just renamed `subscription_send_doc` to `post_subscription` for clarity.

File: test/chw.test.js [hunk 4]
```diff
@@ -782,7 +782,7 @@ describe("app", function() {
                         })
                         .setup.user.addr('27001')
                         .setup.user.state('states_mobile_no')
-                        .input('831112222')
+                        .input('0831112222')
```
Comment (gsvr) [inline test/chw.test.js:785]: @imsickofmaps `83111222` is not actually a valid number input.

**Comment by gsvr** [comment]
@imsickofmaps Please review.

File: test/clinic.test.js [hunk 5]
```diff
@@ -1895,7 +1879,6 @@ describe("app", function() {
                             assert.deepEqual(metrics['test.clinic.avg.sessions_to_register'].values, [5]);
                             assert.deepEqual(metrics['test.clinic.percent_incomplete_registrations'].values, [25]);
                             assert.deepEqual(metrics['test.clinic.percent_complete_registrations'].values, [75]);
-                            assert.deepEqual(metrics['test.sum.subscriptions'].values, [1]);
```
Comment (imsickofmaps) [inline test/clinic.test.js:1898]: Please make an issue for ndoh-control which captures metrics that were removed from go-ndoh that need adding to python so we don't lose any inadvertently

File: src/utils.js (hunk 3 above)
Comment (imsickofmaps) [inline src/utils.js:620]: feels odd to have a function called `post_subscription` that calls an endpoint called `registration` don't you think?

Comment (imsickofmaps) [inline src/utils.js:620]: ignore, just realised this one actually is a subscription! :dizzy_face:

File: src/chw.js (hunk 1 above)
Comment (imsickofmaps) [inline src/chw.js:552]: I think ultimately we need to stop making server side calculated metrics and move to display time calcs for this kind of thing. Fine with this for now.

**Comment by imsickofmaps** [comment]
review done, few minor things

File: test/clinic.test.js (hunk 5 above)
Comment (gsvr) [inline test/clinic.test.js:1898]: I already did (ndoh-control 92)

**Comment by gsvr** [comment]
So :+1: ?

**Comment by imsickofmaps** [comment]
:+1:

**Pull request merged**